src/main/java/org/jvnet/hudson/test/TestPluginManager.java

```diff
@@ -57,6 +66,11 @@
  * @see HudsonTestCase#useLocalPluginManager
  */
 public class TestPluginManager extends PluginManager {
+
+    @CheckForNull
+    private static final boolean DO_NOT_OVERRIDE_BUNDLED_PLUGINS_BY_OLDER_TEST_DEPS =
+            Boolean.getBoolean("jth.pluginManager.doNotOverrideBundledPluginsByOlderTestDeps");
```

Comment:
Maybe -0 on this: I feel it makes the semantics of functional tests a bit harder to understand, and for no very compelling reason: you can already simply ensure that the WAR under test contains sufficiently new versions of the dependencies.